src/App.tsx

```diff
@@ -1,72 +1,94 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback, useRef, useEffect } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu';
+
+
+function debounce(callback: Function, delay: number = 300) {
+  let timerId: number = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+    timerId = window.setTimeout(() => {
+      callback(...args)
+    }, delay)
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [name, setName] = useState('');
+  const [appliedPerson, setAppliedPerson] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const applyPerson = useMemo(() => debounce(setAppliedPerson, 1000), []);
```

Comment:
The debounce delay is hardcoded to 1000ms. Consider making it customizable by passing it as a prop or using a constant.